e2e/puppeteer-custom-environment.ts

```diff
@@ -34,28 +38,31 @@ class PuppeteerCustomEnvironment extends PuppeteerEnvironment {
         const runningTest = state.currentlyRunningTest;
         let testName;
         if (runningTest != null) {
-          testName = state.currentlyRunningTest.name.replace(/\W/g, '-');
+          testName = runningTest.parent.name.replace(/\W/g, '-');
         } else {
           testName = event.test.name.replace(/\W/g, '-');
         }
 
-        const screenshotDir = `logs/screenshot`;
-        const htmlDir = `logs/html`;
-        await fs.ensureDir(screenshotDir);
-        await fs.ensureDir(htmlDir);
+        await fs.ensureDir(this.screenshotDir);
+        await fs.ensureDir(this.htmlDir);
 
         const timestamp = this.localDateTimeString();
-
-        const screenshotFile = `${screenshotDir}/${testName}_${timestamp}.png`;
+        const screenshotFile = `${this.screenshotDir}/${testName}_${timestamp}.png`;
         await this.takeScreenshot(screenshotFile);
-
-        const htmlFile = `${htmlDir}/${testName}_${timestamp}.html`;
+        const htmlFile = `${this.htmlDir}/${testName}_${timestamp}.html`;
         await this.savePageToFile(htmlFile);
+
+        this.failedTest = true;
         break;
       default:
         break;
     }
-
+    if (this.failedTest && event.name === 'test_start') {
+      event.test.mode = 'skip';
```

Comment:
I don't understand what is happening here, please add comments

Reply:
[jest-circus](https://github.com/facebook/jest/blob/master/packages/jest-circus/README.md#overview) is used as jest [runner](https://github.com/all-of-us/workbench/blob/1ab08cbf660ce85d343a64770f9b548e9445a3cd/e2e/jest.config.js#L9). It hooks into test event and state data.
It's fail fast. Because I can't use `bail` option with a custom logger, I have to find another way to stop other tests from running after first failure. If first failure has happened, all following unexecuted tests will be marked as skip.
Comments also added to file.